oat-sa/tao-core-sdk-fe
# Fix/ADF-1166/ESlint errors in tests of utils

### Related to: https://oat-sa.atlassian.net/browse/ADF-1166

### Summary

Fix the ESLint error in the unit tests for `tao-core-utils`.

**Note:** as the changes are pretty heavy, I split them all into several pull requests.

### Details

This pull request only targets the `test/util/*` part of the library.
The full picture can be seen from the main branch which contains all the fixes: https://github.com/oat-sa/tao-core-sdk-fe.git#fix/ADF-1166/eslint-errors

What has been made:
- fix errors that can be automated using `eslint --fix`, then adjust the changes in case by case (mostly string literals)
- fix the wrong annotations (missing/wrong return or param definition)
- fix the more complex cases with some refactoring (rest operator instead of `.apply()`)
- replace `var` by `const` or `let` (but not in all places, only in the touched files when it was not adding too much noise)
- apply object shorthand instead of duplicated function names
- apply arrow functions here and there when it helped, but not systematically to reduce the noise

### How to test
- checkout and install the source code `npm i`
- run the linter: `npm run lint:test`, errors and warnings coming from the `utils` folder should have gone.
- run the unit tests: `npm run test utils`
- install with other repositories, using the following scheme: `"@oat-sa/tao-core-sdk": "https://github.com/oat-sa/tao-core-sdk-fe.git#fix/ADF-1166/eslint-errors",`, then checking the unit tests and playing with TAO
